pkg/monitortests/node/watchnodes/node.go

```diff
@@ -157,6 +148,53 @@ func startNodeMonitoring(ctx context.Context, m monitorapi.RecorderWriter, clien
 			}
 			return intervals
 		},
+		// This function is added to help detect unexpected
+		// node not ready.
+		// We want to fail the monitor test if a node goes not ready
+		// if it is unexpected.
+		// Unexpected in this case means that it went not ready outside
+		// of a MCO config update.
+		func(node, oldNode *corev1.Node) []monitorapi.Interval {
+			var intervals []monitorapi.Interval
+
+			oldConfig := oldNode.Annotations["machineconfiguration.openshift.io/currentConfig"]
+			newDesired := node.Annotations["machineconfiguration.openshift.io/desiredConfig"]
+			isOldNodeReady := isNodeReady(oldNode)
+			isNewNodeReady := isNodeReady(node)
+			isConfigTheSame := oldConfig == newDesired
+			isNodeUnscheduable := node.Spec.Unschedulable
+
+			now := time.Now()
+			if isOldNodeReady && !isNewNodeReady && isConfigTheSame && !isNodeUnscheduable {
+				intervals = append(intervals,
+					monitorapi.NewInterval(monitorapi.SourceUnexpectedReady, monitorapi.Error).
+						Locator(monitorapi.NewLocator().NodeFromName(node.Name)).
+						Message(monitorapi.NewMessage().Reason(monitorapi.NodeUnexpectedReadyReason).
+							HumanMessage("unexpected node not ready")).
+						Display().
+						Build(now, now))
+			}
+			return intervals
+		},
+		func(node, oldNode *corev1.Node) []monitorapi.Interval {
+			var intervals []monitorapi.Interval
+
+			isOldNodeUnReachable := doesNodeHaveUnreachableTaints(oldNode)
+			isNewNodeUnReachable := doesNodeHaveUnreachableTaints(node)
+			isNodeUnscheduable := node.Spec.Unschedulable
+
+			now := time.Now()
+			if !isOldNodeUnReachable && isNewNodeUnReachable && !isNodeUnscheduable {
+				intervals = append(intervals,
+					monitorapi.NewInterval(monitorapi.SourceUnreachable, monitorapi.Warning).
+						Locator(monitorapi.NewLocator().NodeFromName(node.Name)).
+						Message(monitorapi.NewMessage().Reason(monitorapi.NodeUnreachableReason).
```

Comment:
UnexpectedUnreachable.

Reply:
NodeUnexpectedUnreachableReason?

Trying to follow similar naming as the other reasons.

Reply:
```
	NodeUnexpectedReadyReason IntervalReason = "UnexpectedNotReady"
	NodeUnexpectedUnreachableReason     IntervalReason = "UnexpectedUnreachable"

```

Reply:
Yeah, and probably also have a regular unreachable we don't error on